src/parser.cpp

```diff
@@ -0,0 +1,637 @@
+                             */
+                            variable2 = std::make_shared<Variable>();
+                            variable2->setName(variableName);
+                            component2->addVariable(variable2);
+                        } else {
+                            VariableErrorPtr err = std::make_shared<VariableError>();
+                            err->setDescription("Connection variable_2 '" + variableName +
+                                                "' not found in component_2 '" + component2->getName() + "'.");
+                            err->setComponent(component2);
+                            addError(err);
+                            errorOccurred = true;
+                        }
+                    } else {
+                        ModelErrorPtr err = std::make_shared<ModelError>();
+                        err->setDescription("Connection does not name a valid component_2 for connection "
+                                            "between variables variable_1='" + variable1->getName() + "' "
+                                            "and variable_2='" + variableName + "'.");
+                        err->setModel(model);
+                        addError(err);
+                        errorOccurred = true;
+                    }
+                } else {
+                    ModelErrorPtr err = std::make_shared<ModelError>();
+                    err->setDescription("Connection map_variables does not contain a variable_2.");
+                    err->setModel(model);
+                    addError(err);
+                    errorOccurred = true;
+                }
+            } else {
+                ModelErrorPtr err = std::make_shared<ModelError>();
+                err->setDescription("Connection map_variables does not contain a variable_1.");
+                err->setModel(model);
+                addError(err);
+                errorOccurred = true;
+            }
+        } else {
+            ModelErrorPtr err = std::make_shared<ModelError>();
+            err->setDescription("Invalid element type '" + mapVariablesNode->getType() +
+                                "' found in model connection. Expected a map_variables element.");
+            err->setModel(model);
+            addError(err);
+            errorOccurred = true;
+        }
+
+        // Set variable equivalence relationship.
+        if (!errorOccurred) {
+            Variable::addEquivalence(variable1, variable2);
+        }
+        // Continue on to next map_variables if one exists
+        mapVariablesNode = mapVariablesNode->getNext();
+    }
+}
+
+void Parser::loadEncapsulation(const ModelPtr &model, XmlNodePtr &parentComponentNode)
+{
+    while (parentComponentNode) {
+        bool errorOccurred = false;
+        ComponentPtr parentComponent = nullptr;
+        if (parentComponentNode->isType("component_ref")) {
+            if (parentComponentNode->hasAttribute("component")) {
```

Comment:
I think we do probably need to iterate over all present XML nodes and attributes with the parser and log any errors that may come up when building up the objects from input strings. My understanding was that the `Validator` could then be applied to the constructed objects to check whether they meet the Spec. 

So the `Validator` wouldn't actually be touching an input string and the `Parser` is the only place to catch something like a spelling error for an input attribute type.